tests/ui/same_item_push.rs

```diff
@@ -111,4 +113,15 @@ fn main() {
     for _ in 0..10 {
         vec15.push(Box::new(S {}));
     }
+
+    let mut vec16 = Vec::new();
+    for _ in 0..20 {
+        vec16.push(VALUE);
+    }
+
+    let mut vec17 = Vec::new();
+    let item = VALUE;
+    for _ in 0..20 {
+        vec17.push(item);
+    }
```

Comment:
We could add a test (not linted) involving shadowing
```suggestion
    }
    
    let mut vec = Vec::new();
    let item = 42;
    let item = fun();
    for _ in 0..20 {
        vec.push(item);
    }
```

And define the called function outside `main`
```rust
fn fun() -> usize {
    42
}
```